packages/astro/src/env/constants.ts

```diff
@@ -0,0 +1,11 @@
+function resolveVirtualModuleId<T extends string>(id: T): `\0${T}` {
+	return `\0${id}`;
+}
+
+export const VIRTUAL_CLIENT_MODULE_ID = 'astro:env/client';
+export const RESOLVED_VIRTUAL_CLIENT_MODULE_ID = resolveVirtualModuleId(VIRTUAL_CLIENT_MODULE_ID);
```

Comment:
Personally I think it's overkill to have a function that only prepends `\0`. I think it's fine to hardcode it 😅